issue_574_move_skills_section_to_next

KazAttarwala opened the pull request:
# Description

Added softwareSkills component as part of the migration to Next.js.

Fixes #574 

## Type of change

- [X] New feature (non-breaking change which adds functionality)

saadpasta commented on src/pages/softwareSkills.js:
```diff
@@ -0,0 +1,34 @@
+import React from "react";
+import "../styles/softwareSkills.module.scss";
+import {skillsSection} from "../portfolio";
+
+export default function SoftwareSkill() {
+  return (
+    <div>
+      <div className="software-skills-main-div">
+        <ul className="dev-icons">
+          {skillsSection.softwareSkills.map((skills, i) => {
+            return (
+              <li
+                key={i}
+                className="software-skill-inline"
+                name={skills.skillName}
+              >
+                <i className={skills.fontAwesomeClassname}></i>
+                <p>{skills.skillName}</p>
+              </li>
+            );
+          })}
+        </ul>
+      </div>
+    </div>
+  );
+}
+
+// export async function getStaticProps() {
+//   return {
```
No need to create a page inside the pages components you need to create seprate component for this

KazAttarwala replied:
got it working on it right now

saadpasta commented on src/portfolio.js:
```diff
@@ -0,0 +1,397 @@
+  animated: true // Set to false to use static SVG
+};
+
+const greeting = {
+  username: "Saad Pasta",
+  title: "Hi all, I'm Saad",
+  subTitle: emoji(
+    "A passionate Full Stack Software Developer 🚀 having an experience of building Web and Mobile applications with JavaScript / Reactjs / Nodejs / React Native and some other cool libraries and frameworks."
+  ),
+  resumeLink:
+    "https://drive.google.com/file/d/1ofFdKF_mqscH8WvXkSObnVvC9kK7Ldlu/view?usp=sharing", // Set to empty to hide the button
+  displayGreeting: true // Set false to hide this section, defaults to true
+};
+
+// Social Media Links
+
+const socialMediaLinks = {
+  github: "https://github.com/saadpasta",
+  linkedin: "https://www.linkedin.com/in/saadpasta/",
+  gmail: "saadpasta70@gmail.com",
+  gitlab: "https://gitlab.com/saadpasta",
+  facebook: "https://www.facebook.com/saad.pasta7",
+  medium: "https://medium.com/@saadpasta",
+  stackoverflow: "https://stackoverflow.com/users/10422806/saad-pasta",
+  // Instagram, Twitter and Kaggle are also supported in the links!
+  // To customize icons and social links, tweak src/components/SocialMedia
+  display: true // Set true to display this section, defaults to false
+};
+
```
No need to include all this right now

KazAttarwala commented:
@saadpasta the vercel preview keeps failing because react-lottie does not yet support react v18.2.0. Could we update the vercel build settings to do a force npm install?

saadpasta reviewed (commented):
Let's wait #594 to be merged and then you can rebase it and create a new PR for that

saadpasta commented:
@KazAttarwala Can you please rebase your branch and use the latest work

KazAttarwala commented:
@saadpasta I will get this done after the weekend.

saadpasta commented:
@KazAttarwala Waiting for you bro to get this done 🙏

KazAttarwala commented:
@saadpasta Hi Saad, I'm almost done with this PR but I'm getting an error when importing react-lottie and react-reveal. The actual error is `Super expression must either be null or a function, not undefined`. Have you ever encountered a similar issue? I've been trying solutions online but no luck.

muneebahmedayub commented:
@KazAttarwala react reveal has not been updated in years and not working with react 18, @saadpasta i think there should be seperate issue for that and should be done after all other issues

KazAttarwala commented:
@muneebahmedayub @saadpasta the compatibility issue with react 18 occurs for both react-lottie and react-reveal. I tried using lottie-react and react-awesome-reveal instead and they seem to be integrating fine with the current code.

KazAttarwala commented on next.config.js:
```diff
@@ -1,8 +1,36 @@
+    appDir: true
+  },
+  //this is to support automatic camelCasing in module.scss files
+  webpack: config => {
```
this webpack config allows us to reference css classes/ids in camelCase at the component level

KazAttarwala commented on package.json:
```diff
@@ -12,8 +12,22 @@
+    "react-awesome-reveal": "^4.2.3",
+    "react-dom": "18.2.0",
+    "react-easy-emoji": "^1.8.1",
+    "react-lottie": "^1.2.3",
```
I think we can remove react-lottie and react-reveal since they are no longer maintained and incompatible with react 18. The react-awesome-reveal and lottie-react packages seem to be working fine as a replacement.

saadpasta replied:
Yep if that's the case then you can remove it

KazAttarwala commented:
@muneebahmedayub I removed the pages directory because it is reserved for react components, not just exported JS functions. I also moved portfolio.js to the same level as the src directory.

KazAttarwala commented:
@saadpasta should I create a new PR or just leave this one to review?

KazAttarwala commented:
@saadpasta any update on this PR?

saadpasta commented on src/components/SoftwareSkills/SoftwareSkills.jsx:
```diff
@@ -0,0 +1,34 @@
+  );
+}
+
+// export async function getStaticProps() {
+//   return {
+//     props: {
+//       skillsSection
+//     }
+//   }
+// }
```
Remove this comment if no longer needed

saadpasta commented on src/components/displayLottie/DisplayLottie.jsx:
```diff
@@ -0,0 +1,21 @@
+import Loading from "../../containers/loading/Loading";
+
+export default function DisplayLottie({animationData}) {
+  //const animationData = this.props.animationData;
+  // const defaultOptions = {
+  //   loop: true,
+  //   autoplay: true,
+  //   animationData: animationData
+  // };
```
Remove this

saadpasta commented on src/components/displayLottie/DisplayLottie.jsx:
```diff
@@ -0,0 +1,21 @@
+  );
+}
+
+//isClickToPauseDisabled={true} />
```
This one

KazAttarwala reviewed (commented):
@saadpasta should be good to go

KazAttarwala commented:
@saadpasta can you see the icons now? I can see them.

saadpasta commented:
@KazAttarwala LGTM 👍
